lxd/storage/drivers/driver_cephfs_utils.go

```diff
@@ -12,6 +12,12 @@ func (d *cephfs) fsExists(clusterName string, userName string, fsName string) bo
 	return err == nil
 }
 
+// osdPoolExists checks that the Ceph OSD Pool indeed exists.
+func (d *cephfs) osdPoolExists(clusterName string, userName string, osdPoolName string) bool {
+	_, err := shared.RunCommand("ceph", "--name", fmt.Sprintf("client.%s", userName), "--cluster", clusterName, "osd", "pool", "get", osdPoolName, "size")
```

Comment:
I think we need to differentiate between an error ascertaining if the pool exists than from a confirmation that no osd pool exists.

Reply:
We should also update fsExists to equally discriminate between an error and a missing filesystem